CenterForOpenScience/osf.io
# [ENG-1156] Script to consume registration logs

## Purpose

Registrations are going to be saved on the Internet Archive. Part of the registrations are logs, which show the history of the creation of the log.

## Changes

scripts/IA/consume_logs.py - A script to consume the logs as json. Creates a directory of JSON file(s). 

## QA Notes

Ensure all logs appear in the JSON files. Test various page sizes to ensure pagination works correctly

## Documentation

N/A

## Side Effects

N/A

## DevNotes

Usage: `python3 scripts/IA/consume_logs.py -g [GUID] -d [DIRECTORY] -p [PAGESIZE]`. GUID is required. The script will error if one is not provided. Directory is optional. If a directory is not specified, the directory the script is run from will be where the GUID directory is placed. Pagesize is optional. If pagesize is not specified, the default pagesize will be 100. This is useful for testing pagination in JSON files.

## Ticket

https://openscience.atlassian.net/browse/ENG-1156
